src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilderExtensions.cs

```diff
@@ -161,15 +167,20 @@ public static ClusterBuilder ConfigureWithConnectionString(this ClusterBuilder b
             // Server
 
             // Cluster
-            builder = builder.ConfigureCluster(s => s.With(connectionMode: connectionString.Connect));
+#pragma warning disable 618
+            builder = builder.ConfigureCluster(s => s.WithConnection(connectionMode: connectionString.Connect, directConnection: connectionString.DirectConnection));
+#pragma warning restore 618
+
             if (connectionString.Hosts.Count > 0)
             {
                 builder = builder.ConfigureCluster(s => s.With(endPoints: Optional.Enumerable(connectionString.Hosts)));
             }
             if (connectionString.ReplicaSet != null)
             {
                 builder = builder.ConfigureCluster(s => s.With(
+#pragma warning disable 618
```

Comment:
@rstam , can you take a look here, I think these lines can require more changes here

Reply:
Fixed